app/services/skiplagged.js

```diff
@@ -0,0 +1,201 @@
+            req.setTimeout(5000, function() {
+                req.abort();
+            });
+
+            //handles error in request or timeout
+            req.on('error', function(err) {
+                //increment because request has finished with error
+                count++;
+                if(!cb) {
+                    cb = true;
+                    if (err.code === "ECONNRESET" || err.code === "ETIMEDOUT") {
+                        logger.error("Timeout/Connection Reset occured!");
+                    } else {
+                        logger.error("Request Error occured!", err);
+                    }
+                    logger.info('Trying again!\n');
+                    //request not finished => try again
+                    searcher(minLat, minLng, boxSize, delta, callback);
+                }
+            });
+            req.end();
+        }
+    }
+}
+
+function createfunc(j, i, boxSize, delta) {
+    return function(callback) { searcher(j, i, boxSize, delta, callback);};
+}
+module.exports = {
+    search: function() {
+        //initialize variables
+        var funcs = [];
+        var boxSize = 5.0;
+        //experimental: above 0.5 it does not find as much pokemon as below
+        var delta = 0.5;
+
+        //scan for whole world, generates array of functions
+        //currently takes 1-2 hours, also pokeradar doesnt answer requests after like 30 seconds of continous scanning
+        //but then after 20-30 seconds of not answering, responses are again received => idea: switch to proxy
+        if (scanType === 'global') {
+            for (var i = -180.0; i <= 180.0 - boxSize; i = i + boxSize){
+                for (var j = -90.0; j <= 90.0 - boxSize; j = j + boxSize){
+                    funcs.push(createfunc(j, i, boxSize, delta));
+                }
+            }
+        } else if (scanType === 'optimized') {
+            for (var i = 0; i < coords.length; i++){
+                let c = coords[i].split(',');
+                funcs.push(createfunc(parseFloat(c[0]), parseFloat(c[1]), boxSize, delta));
+            }
+        } else {
+            logger.error("Wrong scanType specified!");
+        }
+
+        //executes array of functions in a series (waits for first function to finish and the calls next one ...)
+        //results get stored in pokemon.json file
+        async.series(funcs,
+            function(err, result) {
+                var sum = result.reduce(function (a, b) {return a + b;}, 0);
+                logger.info(sum + ' pokemon found!');
+                if (scanType === 'global') {
+                    fs.writeFile(__base + 'resources/json/skiplaggedCoords.json', JSON.stringify(_.union(coords, coordsFile)), function (err) {
+                        if (err) {
+                            return logger.error(err);
+                        }
+                    });
+                }
+                module.exports.search();
+            });
+    }
+};
```

Comment:
Similar to the other comments, this can also be optimized to serve both pokeRadar and skipLagged, due to similarities.